src/rpc/blockchain.cpp

```diff
@@ -107,7 +107,7 @@ static int ComputeNextBlockAndDepth(const CBlockIndex* tip, const CBlockIndex* b
     return blockindex == tip ? 1 : -1;
 }
 
-UniValue blockheaderToJSON(const CBlockIndex* tip, const CBlockIndex* blockindex)
+UniValue blockheaderToJSON(const CBlockIndex* tip, const CBlockIndex* blockindex) LOCKS_EXCLUDED(cs_main)
```

Comment:
Note that this only does something when compiling with Clang. We should probably also pull in the upstream PR that added `AssertLockNotHeld`.